### PR TITLE
model-data-processing

### PR DESCRIPTION
**Pull request by emmanuelsteven**
In this PR,
- Created model classes for all entities 
- Set up associations between models.
- Created at least one user 
- Created  at least 4 posts written by one of the users
- Create at least 6 posts comments for one of the posts 
- Used other CRUD methods

### CODE REVIEW

**Review by aamirkhan2478 — COMMENTED** [review]
Hi @emmanuelsteven , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: :red_circle: :red_circle: 

You have submitted a project, that hasn't submitted the correct project kindly submit a correct project for you to submit your project validation and specs milestone but I notice there is no validation here and also specs are not there so kindly submit the correct project :slightly_smiling_face: 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  # methods
+
+  def five_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  def update_user_posts_counter
+    author.update(post_counter: author.posts.count)
+  end
```
Comment (elisha2kyakpo1) [inline app/models/post.rb:19]: - [ ] [OPTIONAL] Have you ever considered using the `increment` Ruby method?

__HINT:__ 
```Ruby
def update_user_posts_counter
  author.increment!(:posts_counter)
end
```

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+  # callbacks
+  after_save :update_post_likes_counter
+
+  # methods
+  private
+
+  def update_post_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```
Comment (elisha2kyakpo1) [inline app/models/like.rb:14]: - [ ] [OPTIONAL] Have you ever considered using the `increment` Ruby method?

__HINT:__ 
```Ruby
def update_post_likes_counter
  post.increment!(:likes_counter)
end
```

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @emmanuelsteven, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Model associations are nicely set up👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**